src/core/lib/iomgr/timer_generic.c

```diff
@@ -178,7 +191,7 @@ void grpc_timer_init(grpc_exec_ctx *exec_ctx, grpc_timer *timer,
   GPR_ASSERT(deadline.clock_type == g_clock_type);
   GPR_ASSERT(now.clock_type == g_clock_type);
   timer->closure = closure;
-  timer->deadline = deadline;
+  timer->deadline = timespec_to_atm(deadline);
```

Comment:
Updated to get the rounding right. Agree on this reducing usefulness for high resolution timers: I'm prepared to live with that (and use a different technique for those timers if we need to).

Given epoll_wait() takes a millisecond timeout, I suspect we'll not notice this in practice however.